transhumandesign/kag-base
# Can use tunnel, shops and change class while carrying Dinghy

## Status

- **READY**: this PR is (to the best of your knowledge) ready to be incorporated into the game.

## Description

This fixes [issue #1233](https://github.com/transhumandesign/kag-base/issues/1233).

- Dinghy is set from "medium weight" to "heavy weight" and mass reduced from 1000 to 800. This fixes the inconsistency that Raft had more mass while being considered less weight.
- Disabling action keys is now done in RunnerMovement.as instead of BoatCommon.as. 
It applies not just for Dinghy, but for "heavy weight" tagged things in general now (Dinghy, Raft, Crate with exploding Keg in it).
- You can do one action every 30 ticks, therefore allowing using shops and tunnels, while still disabling slashes/bow.

Applying the key disable for heavy things in general made the most sense to me.
Setting Dinghy to "heavy weight" had to be done, otherwise why disable keys for Dinghy but not for other "medium weight" things like Keg or Boulder?
This also means that Dinghy now reduces your movement speed more, makes heavy wood sounds and may have slightly different logic than before.

Please let me know what you think.
Alternatively, it is also possible to set Dinghy to "medium weight" but then allow action keys again.

## Steps to Test or Reproduce

1. Go to Sandbox.
2. Build three tunnels, some shops and tent.
3. Pick up a Dinghy into your hand.
4. Notice that you can use the tunnels, the shops and change class at tent without problems.
